### PR TITLE
8286671: (fc) Modify sun.nio.ch.FileChannelImpl.map0() to accept a FileDescriptor parameter

### PR DESCRIPTION
**Pull request by bplb**
Add a `FileDescriptor` as the first parameter of `map0()` and remove from the corresponding native code the use of `fd_chan`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286671](https://bugs.openjdk.java.net/browse/JDK-8286671): (fc) Modify sun.nio.ch.FileChannelImpl.map0() to accept a FileDescriptor parameter

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to [1b022d82](https://git.openjdk.java.net/jdk/pull/8689/files/1b022d82c97fe9d5e4daa33e3f513df25757f614)
 * [Jaikiran Pai](https://openjdk.java.net/census#jpai) (@jaikiran - Committer) ⚠️ Review applies to [1b022d82](https://git.openjdk.java.net/jdk/pull/8689/files/1b022d82c97fe9d5e4daa33e3f513df25757f614)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8689/head:pull/8689` \
`$ git checkout pull/8689`

Update a local copy of the PR: \
`$ git checkout pull/8689` \
`$ git pull https://git.openjdk.java.net/jdk pull/8689/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8689`

View PR using the GUI difftool: \
`$ git pr show -t 8689`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8689.diff">https://git.openjdk.java.net/jdk/pull/8689.diff</a>

</details>

### CODE REVIEW

File: src/java.base/unix/native/libnio/ch/FileChannelImpl.c [hunk 1]
```diff
@@ -50,13 +50,10 @@
 #include "java_lang_Integer.h"
 #include <assert.h>
 
-static jfieldID chan_fd;        /* jobject 'fd' in sun.nio.ch.FileChannelImpl */
-
 JNIEXPORT jlong JNICALL
 Java_sun_nio_ch_FileChannelImpl_initIDs(JNIEnv *env, jclass clazz)
```
Comment (jaikiran) [inline src/java.base/unix/native/libnio/ch/FileChannelImpl.c:54]: Hello Brian, now that this function doesn't initialize ids, should it instead be renamed to something else? Also the corresponding `initIDs` method on the Java side? And perhaps remove this comment from `FileChannelImpl.java`:

```
    // Caches fieldIDs
    private static native long initIDs();
```

File: src/java.base/windows/native/libnio/ch/FileChannelImpl.c [hunk 2]
```diff
@@ -36,8 +36,6 @@
 #include <Mswsock.h>
 #pragma comment(lib, "Mswsock.lib")
 
-static jfieldID chan_fd; /* id for jobject 'fd' in java.io.FileChannel */
-
 /**************************************************************
  * static method to store field ID's in initializers
  * and retrieve the allocation granularity
```
Comment (jaikiran) [inline src/java.base/windows/native/libnio/ch/FileChannelImpl.c:41]: Should we remove the "to store field ID's in initializers" part from this comment?

**Comment by jaikiran** [comment]
The copyright year on the Windows variant of `FileChannelImpl.c` will need an update.

**Review by AlanBateman — APPROVED** [review]
Thanks for doing this cleanup, it removes a long standing inconsistency. I agree with Jai on removing the obsolete/confusion comment and I assume you'll do that before integrating.

File: src/java.base/unix/native/libnio/ch/FileChannelImpl.c (hunk 1 above)
Comment (bplb) [reply src/java.base/unix/native/libnio/ch/FileChannelImpl.c:54]: Yeah I had thought of that as well but shined it on. I'll update before integrating.

**Review by jaikiran — APPROVED** [review]

**Comment by bplb** [comment]
> Thanks for doing this cleanup, it removes a long standing inconsistency. I agree with Jai on removing the obsolete/confusion comment and I assume you'll do that before integrating.

Fixed by 0c707e41492a682967f522ce03894f87a133a16f.

**Comment by bplb** [comment]
> The copyright year on the Windows variant of `FileChannelImpl.c` will need an update.

Fixed by 0c707e41492a682967f522ce03894f87a133a16f.

File: src/java.base/windows/native/libnio/ch/FileChannelImpl.c (hunk 2 above)
Comment (bplb) [reply src/java.base/windows/native/libnio/ch/FileChannelImpl.c]: Removed in 0c707e41492a682967f522ce03894f87a133a16f.

**Pull request closed without merging**